src/lib.rs

```diff
@@ -162,7 +165,8 @@ impl AsyncDb {
     pub fn add_data<T>(&self, measure: T) -> AddData
         where T: Measurement
     {
-        let mut request = client::Request::new(hyper::Method::Post, self.write_endpoint.clone());
+        let mut request = client::Request::new(
+            hyper::Method::Post, hyper::Uri::from_str(self.write_endpoint.as_str()).unwrap());
```

Comment:
It's unusual to use `from_str` directly; why did you choose that instead of the idiomatic `self.write_endpoint.as_str().parse()`?